webapp/security/views.py

```diff
@@ -333,5 +342,208 @@ def cve(cve_id):
 # ===
 
 
-def api_create_cve():
-    pass
+def create_cve():
+    """
+    Receives a POST request from load_cve.py
+    Parses the object and bulk inserts with add_all()
+    @params json: the body of the request
+    """
+
+    data = flask.request.get_json()
+    response = flask.jsonify({"message": "Unable to get body"}), 400
+    packages = []
+    references = []
+    bugs = []
+
+    # Check if CVE exists by candidate
+    if db_session.query(CVE).filter(CVE.id == data["id"]).count() > 0:
+        response = flask.jsonify({"message": "CVE already exists"}), 400
```

Comment:
Presumably if a CVE already exists then `db_session.commit()` will fail below. Couldn't you roll this response into the `except` statement down there?

That way you avoid doing this extra query, and you simplify the code.

Reply:
By the way, I think this stackoverflow answer makes a convincing case for a `409` status in this case:

> My feeling is 409 Conflict is the most appropriate, however, seldom seen in the wild of course:
> 
> > The request could not be completed due to a conflict with the current state of the resource. This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request. The response body SHOULD include enough information for the user to recognize the source of the conflict. Ideally, the response entity would include enough information for the user or user agent to fix the problem; however, that might not be possible and is not required.
> > 
> > Conflicts are most likely to occur in response to a PUT request. For example, if versioning were being used and the entity being PUT included changes to a resource which conflict with those made by an earlier (third-party) request, the server might use the 409 response to indicate that it can't complete the request. In this case, the response entity would likely contain a list of the differences between the two versions in a format defined by the response Content-Type.

https://stackoverflow.com/a/3826024/613540